src/foremast/pipeline/create_pipeline_s3.py

```diff
@@ -0,0 +1,144 @@
+#   Foremast - Pipeline Tooling
+#
+#   Copyright 2016 Gogo, LLC
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Create Pipelines for Spinnaker."""
+import collections
+import json
+import logging
+import os
+from pprint import pformat
+
+import requests
+
+from ..consts import API_URL
+from ..exceptions import SpinnakerPipelineCreationFailed
+from ..utils import get_details, get_properties, get_subnets, get_template
+from .clean_pipelines import clean_pipelines
+from .construct_pipeline_block_s3 import construct_pipeline_block_s3
+from .create_pipeline import SpinnakerPipeline
+from .renumerate_stages import renumerate_stages
+
+
+class SpinnakerPipelineS3(SpinnakerPipeline):
+    """Manipulate Spinnaker Pipelines.
+
+    Args:
+        app (str): Application name.
+        trigger_job (str): Jenkins trigger job.
+        base (str): Base image name (i.e: fedora).
+        prop_path (str): Path to the raw.properties.json.
+    """
+
+    def render_wrapper(self, region='us-east-1'):
```

Comment:
We should abstract this out, it seems like we are duplicating code with the other pipeline types. I think it is fine for this PR but worth doing in the future.